src/py_pecg/filtering_module.py

```diff
@@ -0,0 +1,564 @@
+    RL = int(r * Lx)
+    RN = int(r * n)
+
+    # Design the linear-phase FIR filter of type I
+    b = designInterpFilt(r, n, cutoff)
+
+    # Use the filter B to perform the interpolation
+    odt = np.zeros_like(xCol[0]).flatten()
+    yCol = np.zeros((RL, 1), dtype=odt.dtype).flatten()
+    temp_indices = np.arange(
+        0, RL, int(r)
+    )  # Create indices 0, r, 2r, ..., up to RL-1
+    yCol[temp_indices] = xCol
+
+    # Filter a fabricated section of data
+    od = np.zeros((2 * RN, 1)).flatten()
+    temp_indices_2 = np.arange(0, RN * 2, int(r))
+    temp_indices_2_B = np.arange(2 * int(n), 0, -1)
+    od[temp_indices_2] = 2 * xCol[0] - xCol[temp_indices_2_B]
+    zi = lfilter(np.flip(b), 1, np.flip(od))
+    zi = np.flip(zi)
+
+    yCol, zf = lfilter(b, 1, yCol, zi=zi)
+    temp_indices_3 = np.arange(0, (Lx - int(n)) * int(r))
+    temp_indices_3_B = np.arange(RN, RL)
+    yCol[temp_indices_3] = yCol[temp_indices_3_B]
+
+    # Make sure right hand points of data have been correctly interpolated
+    od = np.zeros((2 * RN, 1)).flatten()
+    temp_indices_4 = np.arange(0, RN * 2, int(r))
+    temp_indices_4_B = np.arange(Lx - 2, Lx - 2 * int(n) - 2, -1)
+    od[temp_indices_4] = 2 * xCol[Lx - 1] - (xCol[temp_indices_4_B])
+    od, _ = lfilter(b, 1, od, zi=zf)
+
+    temp_indices_5 = np.arange(RL - RN, RL)
+    temp_indices_5B = np.arange(0, RN)
+    yCol[temp_indices_5] = od[temp_indices_5B]
+    # yCol[RL-RN+1:RL,1] = od[1:RN,1];
+
+    # Convert output to be a row vector if the input is a row
+    if idataIsRow:
+        odata = np.reshape(yCol, (1, RL))
+    else:
+        odata = yCol
+
+    return [odata.flatten(), b]
+
+
+"""
+Credits:
+    Original authors: Juan Pablo Martínez, Rute Almeida, Salvador Olmos,
+    Ana Paula Rocha, Pablo Laguna
+    Original Publication: A wavelet-based ECG delineator:
+    evaluation on standard databases
+    Original DOI: 10.1109/TBME.2003.821031.
+"""
+
+
+def quadratic_splines_filterbank(fs: float, messages: dict):
+    """
+    Obtains the quadratic splines wavelet filterbank filters from
+    scale 1 to 4 as a function of the sampling frequency, in order to use filters
+    with similar analog frequency behaviour for diferent sampling frecuencies.
+
+    Args:
+        fs (float): Sampling frequency of the target signal.
+        messages (dict): A dictionary of in-code configurations.
+
+    Returns:
+        quadratic_filter_array (list): A list of numpy arrays representing filters
+        at different scales.
+        messages (dict): Dictionary containing information relevant to the algorithm.
+
+    Note:
+        Initializing algorithme a trous (Decimation)
+        Transfer Functions:
+        Q1(w) = G(w);
+        Q2(w) = H(w)G(2w);
+        Q3(w) = H(w)H(2w)G(4w);
+        Q4(w) = H(w)H(2w)H(4w)G(8w);
+        Q5(w) = H(w)H(2w)H(4w)H(8w)G(16w);
+    """
+    if fs < 0:
+        raise ValueError("Sampling Frequency must be positive, and non-zero.")
```

Comment:
```suggestion
    if fs <= 0:
        raise ValueError("Sampling Frequency must be positive, and non-zero.")
```
You said it should be non-zero